### DISCUSSION TITLE
Can we use new Observation API instead ObservableObject in the future version?

### DISCUSSION THREAD

**Discussion by AreroKetahi** [sparkle-project/Sparkle#2424] [Q&A]
Hello! I'm noticed that in the version 2.4.2 (even in latest beta version) Sparkle, we cannot use the new `Observation` API which release at WWDC23 to instead elder `ObservableObject` API. 
For my perspective, the new `Observation` API can greatly simplified code. Could this suggestion be implemented in a future release? I'm looking forward to it!

**Comment by zorgiepoo** [comment]
I don't know anything about this. Please elaborate on how you'd use this new API, which Sparkle APIs does this pertain to (at least one example), and how you think Sparkle would be able to support this. Is this a Swift API or a SwiftUI API?

**Comment by zorgiepoo** [comment]
It looks like these are SwiftUI constructs. Our current [sample/documentation code](https://sparkle-project.org/documentation/programmatic-setup/) uses `ObservableObject` as far as I see. The framework only uses/exposes Obj-C currently so I assume this is a documentation issue. However our documentation is intended to target developers that may be supporting older deployment targets, so it might not be right to show off a newer API (I have not looked to see if it would work).

**Reply by AreroKetahi** [reply]
I do understand that app should support lower os version, thanks to your answering!
